runbot_gitlab_minimal/models/runbot_repo.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Copyright <2017> <Vauxoo info@vauxoo.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import os
+import logging
+import re
+import urllib
+
+import requests
+
+from openerp import models, fields
+
+_logger = logging.getLogger(__name__)
+
+
+def custom_build(func):
+    """Decorator for functions which should be overwritten only if
+    uses_gitlab is enabled in repo.
+    """
+    def custom_func(self, cr, uid, ids, *args, **kwargs):
+        custom_ids = []
+        regular_ids = []
+        ret = None
+        for repo in ids:
+            if repo.uses_gitlab:
+                custom_ids.append(repo.id)
+            else:
+                regular_ids.append(repo.id)
+            custom = self.browse(cr, uid, custom_ids)
```

Comment:
A browse is not required since that we are using old api where a list of integers is received